stdio/fprintf.c

```diff
@@ -59,38 +98,73 @@ int fprintf(FILE *file, const char *format, ...)
 
 int vfprintf(FILE *file, const char *format, va_list arg)
 {
-	fprintf_ctx_t ctx;
+	struct feed_ctx_s ctx;
+	size_t res;
+
+	ctx.h.file = file;
+	ctx.htype = 0;
+	ctx.n = 0;
+	ctx.total = 0;
+	ctx.error = 0;
+
+	format_parse(&ctx, format_feed, format, arg);
+
+	if (ctx.n != 0) {
+		res = fwrite(ctx.buff, 1, ctx.n, ctx.h.file);
+		ctx.total += res;
+
+		if (ctx.n != res) {
+			ctx.error = -1;
+		}
+	}
+
+	if (ctx.error < 0) {
+		return -1; /* check error with feof() or ferror() */
+	}
+
+	return ctx.total;
```

Comment:
Less lines, avoids early return easily 
```suggestion
	return (ctx.error < 0) ? -1 /* check error with feof() or ferror() */ : ctx.total;
```